platforms/jvm/language-java/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadIntegrationTest.groovy

```diff
@@ -70,7 +72,7 @@ class JavaToolchainDownloadIntegrationTest extends AbstractIntegrationSpec {
     }
 
     def 'toolchain selection that requires downloading fails when it is disabled'() {
-        setFoojayDiscoToolchainProvider()
+        settingsFile << """${applyToolchainResolverPlugin("CustomToolchainResolver", noUrlResolverCode())}"""
```

Comment:
💅 Consider extracting it into a helper method like before.

Reply:
writing directly into the setting file, just like writing into the build script file keeps things consistent and at the same level of abstraction; I would do the helper method if I would rewrite the whole test